testsuite/shared/src/main/java/org/jboss/as/test/shared/TestLogHandlerSetupTask.java

```diff
@@ -0,0 +1,121 @@
+ */
+
+package org.jboss.as.test.shared;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.as.test.shared.util.LoggingUtil;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+
+/**
+ * Used to search for strings in logs, it adds a handler so the search is done on a small file
+ * @author tmiyar
+ *
+ */
+public abstract class TestLogHandlerSetupTask implements ServerSetupTask {
+
+    private final Logger LOGGER = Logger.getLogger(TestLogHandlerSetupTask.class);
+
+    @Override
+    public void setup(ManagementClient managementClient, String containerId) throws Exception {
+
+        //add handler
+        ModelNode addTestLogOp = org.jboss.as.controller.operations.common.Util.createAddOperation(PathAddress.pathAddress().append(SUBSYSTEM, "logging")
+                .append("periodic-rotating-file-handler", getHandlerName()));
+        addTestLogOp.get("level").set(getLevel());
+        addTestLogOp.get("append").set("true");
+        addTestLogOp.get("suffix").set(".yyyy-MM-dd");
+        ModelNode file = new ModelNode();
+        file.get("relative-to").set("jboss.server.log.dir");
+        file.get("path").set(getLogFileName());
+        addTestLogOp.get("file").set(file);
+        addTestLogOp.get("formatter").set("%-5p [%c] (%t) %s%e%n");
+        applyUpdate(managementClient.getControllerClient(), addTestLogOp, false);
+
+        //add category with new handler
+        Collection<String> categories = getCategories();
+        if (categories == null || categories.isEmpty()) {
+            LOGGER.warn("getCategories() returned empty collection.");
+            return;
+        }
+
+        for (String category : categories) {
+            if (category == null || category.length() == 0) {
+                LOGGER.warn("Empty category name provided.");
+                continue;
+            }
+            ModelNode op = org.jboss.as.controller.operations.common.Util.createAddOperation(PathAddress.pathAddress().append(SUBSYSTEM, "logging").append("logger", category));
+            op.get("level").set(getLevel());
+            op.get("handlers").add(getHandlerName());
+            applyUpdate(managementClient.getControllerClient(), op, false);
+        }
+    }
+
+    @Override
+    public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+        // remove file
+        Path logPath = LoggingUtil.getLogPath(managementClient, "periodic-rotating-file-handler", getHandlerName());
+        Files.deleteIfExists(logPath);
```

Comment:
The actual delete should happen at the end as WIndows can be very picky about files being logged. So the remove would need to happen first.